posthog-android/build.gradle.kts

```diff
@@ -89,6 +89,7 @@ dependencies {
     implementation("androidx.lifecycle:lifecycle-common-java8:${PosthogBuildConfig.Dependencies.LIFECYCLE}")
     implementation("androidx.core:core:${PosthogBuildConfig.Dependencies.ANDROIDX_CORE}")
     implementation("com.squareup.curtains:curtains:${PosthogBuildConfig.Dependencies.CURTAINS}")
+    implementation("androidx.compose.ui:ui-android:1.7.3")
```

Comment:
I think we should target the minimal version that has all the methods we need such as:

> import androidx.compose.ui.Modifier
> import androidx.compose.ui.semantics.SemanticsPropertyKey
> import androidx.compose.ui.semantics.semantics
> import androidx.compose.ui.node.RootForTest
> import androidx.compose.ui.semantics.SemanticsProperties
> import androidx.compose.ui.semantics.getAllSemanticsNodes
> import androidx.compose.ui.semantics.getOrNull

Dependency resolution will use the latest version anyway, but if we use a new one that breaks something and people depend on an older one, it's gonna be our SDK doing this.

Reply:
Another option is to make this dependency `compileOnly`, and we check if its in the classpath before using its classes, probably with reflection as well (Class.forName)

Reply:
Unless we need something from `ui-android`, we could depend on the `androidx.compose.ui:ui`

Reply:
the `compose` dependency is still `implementation` and not `compileOnly` so we still bring the dependency even if not using the modifier, have you tried to fix that?

Reply:
ah yes, I think we can simply change that to `compileOnly`

Reply:
try to run the sample without any androidx.compose libs and enable the session reply just to be sure its compatible if someone isn't using compose at all, it should not crash or anything, if that works, we are gold.

Reply:
I'll do it 👍

Reply:
just tried it right now. everything working normally :)

Reply:
cool, i will run some tests tomorrow morning before merging and if all ok, that is DONE yay thanks for contributing.